### PR TITLE
add support for mob

### PR DESCRIPTION
**Pull request by sumny**
`mob`: somewhat experimental
- the fitting function (i.e., model to be used for partitioning) is passed via the parameter `fit` and the formula's right hand side must be specified via `rhs` as described in `partykit::mob`.
- an appropriate prediction function must be specified, passed via `predict_fun`, because a `mob` learner is too general for a "hardcoded" predict method; see `?predict.modelparty` and the description in their vignette. See also the minimal examples in the tests
- all in all this learner should also be better explained, maybe a gallery post because this is not really straightforward to set up.

### CODE REVIEW

**Comment by sumny** [comment]
@pat-s this is now ready for review (checklist passed and CI and paramtest as well; expect for macOS devel seems to be broken)

File: README.md [hunk 1]
```diff
@@ -2,14 +2,13 @@
 
 <!-- badges: start -->
 
-[![R CMD Check via {tic}](https://github.com/mlr3learners/mlr3learners.partykit/workflows/R%20CMD%20Check%20via%20{tic}/badge.svg?branch=master)](https://github.com/mlr3learners/mlr3learners.partykit/actions)
```
Comment (pat-s) [inline README.md:5]: Was this by accident?

File: R/zzz.R [hunk 2]
```diff
@@ -13,15 +13,19 @@ register_mlr3 = function(libname, pkgname) {
...
+.onLoad = function(libname, pkgname) {
+  # nolint
   register_mlr3()
   setHook(packageEvent("mlr3", "onLoad"), function(...) register_mlr3(),
     action = "append")
 }
 
-.onUnload = function(libpath) { # nolint
+.onUnload = function(libpath) {
+  # nolint
```
Comment (pat-s) [inline R/zzz.R:28]: Watch the linebreaking, probably introduced by styler. The issue is already fixed -> update {styler} and correct the placing of `# nolint`.

File: tests/testthat/test_classif_mob.R [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    glm(y ~ 1, family = binomial, start = start, ...)
+  }
+  learner = LearnerClassifMob$new()
+  learner$param_set$values$rhs = " . "
+  learner$param_set$values$fit = logit_
+  learner$param_set$values$additional = list(maxit = 100)
+  learner$feature_types = c("logical", "integer", "numeric", "factor", "ordered")
+  learner$properties = c("twoclass", "weights")
+
+  # FIXME: this is from mlr3learners helpers
```
Comment (pat-s) [inline tests/testthat/test_classif_mob.R:15]: We should move this to the autotest helpers in mlr3 with a short comment. Can you please open a PR there?

**Review by pat-s — CHANGES_REQUESTED** [review]
Thanks! Yes macOS devel is broken atm.

Please change the following points:

- [x] Be explicit about imports, i.e. use `mlr3misc::invoke()` (I know that the Ctree learner also lacks this)


- [x] Fix the lintr exclusion as suggested in the comment.

- [x] Add yourself as an auth to DESC

- [x] After approval, bump the version and do a [drat release](https://github.com/mlr3learners/mlr3learners.drat#how-to-add-a-mlr3learner-to-mlr3learnersdrat) (LMK if you face troubles)

File: tests/testthat/test_classif_mob.R (hunk 3 above)
Comment (sumny) [reply tests/testthat/test_classif_mob.R:15]: done, mlr-org/mlr3#486

**Comment by sumny** [comment]
>     * [x]  After approval, bump the version and do a [drat release](https://github.com/mlr3learners/mlr3learners.drat#how-to-add-a-mlr3learner-to-mlr3learnersdrat) (LMK if you face troubles)

I just did this - seems like it worked as expected, right?

**Pull request merged**